tools/splat_ext/pm_sbn.py

```diff
@@ -0,0 +1,886 @@
+
+
+        initHeader = INITHeader(
+            size = init_file_end - init_header_offset,
+
+            tblOffset = song_entry_offset - init_header_offset,
+            tblSize = song_entry_end - song_entry_offset,
+
+            entriesOffset = bank_entry_offset - init_header_offset,
+            entriesSize = bank_entry_end - bank_entry_offset,
+
+            shortsOffset = mseq_offset - init_header_offset,
+            shortsSize = mseq_end - mseq_offset,
+        )
+
+        struct.pack_into(INITHeader.fstring, data, init_header_offset, *initHeader)
+
+        current_bank_offset = bank_entry_offset
+        for bank in self.init.bk_entries:
+            struct.pack_into(BufferEntry.fstring, data, current_bank_offset, *bank)
+            current_bank_offset += BufferEntry.length
+
+        sentinel = BufferEntry(0xffff, 0x00, 0x00)
+        struct.pack_into(BufferEntry.fstring, data, current_bank_offset, *sentinel)
+
+        current_song_offset = song_entry_offset
+        for song in self.init.song_entries:
+            struct.pack_into(InitSongEntry.fstring, data, current_song_offset, *song)
+            current_song_offset += InitSongEntry.length
+
+        sentinel = InitSongEntry(0xffff,0x00,0x00,0x00)
+        struct.pack_into(InitSongEntry.fstring, data, current_song_offset, *sentinel)
+
+        current_mseq_offset = mseq_offset
+        for mseq in self.init.mseq_entries:
+            struct.pack_into(">H", data, current_mseq_offset, mseq)
+            current_mseq_offset += 2
+        struct.pack_into(">H", data, current_mseq_offset, 0xffff)
+
+        return bytes(data)
+
+    # Write the files in the SBN to a directory.
+    def write(self, path: Path):
+        if not path.exists():
+            path.mkdir()
+        
+        with open("unknown.bin", "wb") as f:
+            f.write(self.unknown_data)
+
+        for sbn_file in self.files:
+            try:
+                filename = sbn_file.file_name()
+                sbn_file.write(path / filename)
+            except Exception as e:
+                raise Exception(f"Failed to write {sbn_file}: {e}")
+
+        with open(path / "sbn.yaml", "w") as f:
+            # Filename->ID map
+            f.write("# Mapping of filenames to entry IDs. Use 'id: auto' to automatically assign a unique ID.\n")
+            f.write("""
+# 'fakesize is an interesting case. In the final ROM, the size of a file is stored in the file header and the entry table. 
+# For bank files, however, the entry table has seemingly random data stored for file size. This data appears to be unused.
+# Rather than figure out how this data is actually generated, I hardcoded it. What does this mean for modders?
+# Well, for non bank files and bankfiles less than 65536 bytes it means nothing. Simply don't set the fakesize field and it will be fine. 
+# As for bank files that are more than 65536 bytes long, you have to set fakesize to something but it doesn't appear to matter what.\n""")
```

Comment:
> What does this mean for modders?

Generally, as long as it works, modders aren't bothered about details such as this.

I figure we should treat `fakesize` as a boolean: 
- when size < 65536 bytes, fakesize defaults to 0
- when size >= 65536 bytes, fakesize defailts to 1